### PR TITLE
Add html parsers

### PR DESCRIPTION
**Pull request by eyurtsev**
# Add bs4 html parser

* Some minor refactors
* Extract the bs4 html parsing code from the bs html loader
* Move some tests from integration tests to unit tests

### CODE REVIEW

File: langchain/document_loaders/parsers/html/bs4.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+logger = logging.getLogger(__name__)
+
+
+class BS4HTMLParser(BaseBlobParser):
+    """Parser that uses beautiful soup to parse HTML files."""
+
+    def __init__(
+        self,
+        features: str = "lxml",
+        bs_kwargs: Optional[Mapping[str, Any]] = None,
```
Comment (dev2049) [inline langchain/document_loaders/parsers/html/bs4.py:19]: ooc (out of curiosity) how do you think about passing in kwargs as dict vs **kwargs?

**Review by dev2049 — APPROVED** [review]
🥳

File: langchain/document_loaders/parsers/html/bs4.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class BS4HTMLParser(BaseBlobParser):
+    """Parser that uses beautiful soup to parse HTML files."""
+
+    def __init__(
+        self,
+        features: str = "lxml",
```
Comment (dev2049) [inline langchain/document_loaders/parsers/html/bs4.py:18]: should we add this to bs4html loader as well?

File: langchain/document_loaders/parsers/html/bs4.py (hunk 1 above)
Comment (eyurtsev) [reply langchain/document_loaders/parsers/html/bs4.py:19]: why not :)

My main concern tends to be that there might be another function that'll require them... but here maybe it's OK to assume it goes into the initializer by default, and everything else has to be explicitly specified

File: langchain/document_loaders/parsers/html/bs4.py (hunk 2 above)
Comment (eyurtsev) [reply langchain/document_loaders/parsers/html/bs4.py:19]: it's already there, just hidden in the init

**Pull request merged**